tests/unit/Hooks/ImageHooksTest.php

```diff
@@ -202,10 +142,61 @@ protected function mockJpgFileMethods($node, $path) {
 	 * @param $path
 	 * @param $dimensions
 	 */
-	protected function mockGetImageSize($mockClass, $path, $dimensions) {
-		$mockClass->expects($this->once())
+	//TODO: This is still not correctly implemented
+	protected function mockGetImageSize($path, $dimensions) {
+		$this->imagehooks->expects($this->once())
 			->method('getimagesize')
 			->with($path)
 			->willReturn($dimensions);
 	}
+
+	/**
+	 * @return object|\PHPUnit_Framework_MockObject_MockObject
+	 */
+	private function mockImageDimensionMapper() {
+		/*return new ImageDimensionMapper(
+			$this->db
+		);*/
+
+		return $this->getMockBuilder('OCA\MediaMetadata\Services\ImageDimensionMapper')
+			->setConstructorArgs($this->db)
+			->getMock();
+
+		/*
+		 * Another Method
+		 * $app = new \OCA\MediaMetadata\AppInfo\Application();
+		 * $container = $app->getContainer();
+		 *
+		 * return new ImageDimensionMapper(
+		 *     $container->query('ServerContainer')->getDb()
+		 * );
+		 */
+	}
+
+	/**
+	 * @return object|\PHPUnit_Framework_MockObject_MockObject
+	 */
+	private function mockImageDimension() {
+		//return new ImageDimension();
+
+		return $this->getMockBuilder('OCA\MediaMetadata\Services\ImageDimension')
+			->disableOriginalConstructor()
+			->getMock();
+	}
+
+	/**
+	 * @param array $mockedMethods
+	 * @return object|ImageHooks|\PHPUnit_Framework_MockObject_MockObject
+	 */
+	private function mockPostCreate() {
```

Comment:
There is a misunderstanding. Your previous method was doing 2 things, so you needed to put one part in setUp so that you could call the methods to test (the example I gave you from Gallery) and one part in a postCreate mock method so that it could be called in the test of register.
What you did is to put the part which was destined for setUp in the postCreate mock method.
There is no need to create getImageHooks. It's easier to read when we have everything for the main object creation in the setUp method, like in the Gallery example.